include/openthread/platform/radio.h

```diff
@@ -618,6 +621,21 @@ extern void otPlatRadioTxDone(otInstance *aInstance, otRadioFrame *aFrame, otRad
  */
 extern void otPlatDiagRadioTransmitDone(otInstance *aInstance, otRadioFrame *aFrame, otError aError);
 
+/**
+ * The radio driver calls this method to notify OpenThread to process transmit security for the frame,
+ * this happens when the frame includes Header IE(s) that were updated before transmission.
+ *
+ * This function is used when feature `OPENTHREAD_CONFIG_MAC_HEADER_IE_SUPPORT` is enabled.
+ *
+ * @note This function can be called from interrupt context and it would only read/write data passed in
+ *       via @p aFrame, but would not read/write any state within OpenThread.
+ *
+ * @param[in]  aInstance   The OpenThread instance structure.
+ * @param[in]  aFrame      The radio frame which needs to process transmit security.
+ *
+ */
+extern void otPlatRadioFrameUpdated(otInstance *aInstance, otRadioFrame *aFrame);
```

Comment:
We intentionally removed the callback in #4318 to avoid this callback accessing OpenThread state.